lecture/src/test/java/de/tum/in/www1/artemis/lecture/LectureIntegrationTest.java

```diff
@@ -0,0 +1,371 @@
+    public void testAll_asStudent() throws Exception {
+        this.testAllPreAuthorize();
+    }
+
+    @AfterEach
+    public void resetDatabase() {
+        database.resetDatabase();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void createLecture_correctRequestBody_shouldCreateLecture() throws Exception {
+        Course course = courseRepository.findByIdElseThrow(this.course1.getId());
+
+        Lecture lecture = new Lecture();
+        lecture.title("loremIpsum");
+        lecture.setCourse(course);
+        lecture.setDescription("loremIpsum");
+        Lecture returnedLecture = request.postWithResponseBody("/api/lectures", lecture, Lecture.class, HttpStatus.CREATED);
+
+        assertThat(returnedLecture).isNotNull();
+        assertThat(returnedLecture.getId()).isNotNull();
+        assertThat(returnedLecture.getTitle()).isEqualTo(lecture.getTitle());
+        assertThat(returnedLecture.getCourse().getId()).isEqualTo(lecture.getCourse().getId());
+        assertThat(returnedLecture.getDescription()).isEqualTo(lecture.getDescription());
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void createLecture_alreadyId_shouldReturnBadRequest() throws Exception {
+        Lecture lecture = new Lecture();
+        lecture.setId(1L);
+        request.postWithResponseBody("/api/lectures", lecture, Lecture.class, HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void updateLecture_correctRequestBody_shouldUpdateLecture() throws Exception {
+        Lecture originalLecture = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoals(lecture1.getId()).get();
+        originalLecture.setTitle("Updated");
+        originalLecture.setDescription("Updated");
+        Lecture updatedLecture = request.putWithResponseBody("/api/lectures", originalLecture, Lecture.class, HttpStatus.OK);
+        assertThat(updatedLecture.getTitle()).isEqualTo("Updated");
+        assertThat(updatedLecture.getDescription()).isEqualTo("Updated");
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void updateLecture_NoId_shouldReturnBadRequest() throws Exception {
+        Lecture originalLecture = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoals(lecture1.getId()).get();
+        originalLecture.setId(null);
+        request.putWithResponseBody("/api/lectures", originalLecture, Lecture.class, HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void getLectureForCourse_withOutLectureUnits_shouldGetLecturesWithOutLectureUnits() throws Exception {
+        List<Lecture> returnedLectures = request.getList("/api/courses/" + course1.getId() + "/lectures", HttpStatus.OK, Lecture.class);
+        assertThat(returnedLectures.size()).isEqualTo(2);
```

Comment:
You could do quite some assertion improvements like
.size()).isEqual() => ).hasSize()

or moving long assertions parameter chains in their own variables.
.isEmpty()).isTrue() => ).isEmpty()

However, if you consider this to be out of scope and you prefer not do do these changes here, that's okay too.

Reply:
changed it